src/status_im2/common/bottom_sheet/view.cljs

```diff
@@ -12,11 +12,11 @@
             [react-native.safe-area :as safe-area]
             [reagent.core :as reagent]))
 
-(def bottom-sheet-js (js/require "../src/js/bottom_sheet.js"))
+(def ^:private bottom-sheet-js (js/require "../src/js/bottom_sheet.js"))
```

Comment:
could you elaborate on `^:private` whats the point of using it ?

Reply:
but why? i mean I've never seen any issues with that , it's just a noise in the code all these private stuff , like `name-` or `private`

Reply:
My two cents: I used to use `^:private` and `name-` for a while. But I didn't see any benefits from it, and got tired of the noise and extra syntax in testing them, so I stopped doing it. I suspect that in FP (as opposed to OOP), it's less valuable, except for when you want to hide implementation details from external consumers (when you have them). I totally wouldn't mind going back to following this practice, if that's what we're doing, if I knew what the benefits were. I did notice that the clojure style guide says it is good to make private functions private, but doesn't talk about when it good to make those parts private. Maybe this is too small a thing to bother discussing the merits of, which is fine too.

Reply:
yeah i agree, lets discuss, and add this into our guidelines cc @ilmotta

Reply:
Good discussion! Yeah, I agree it's currently an open point in the team. I will try to shed some light to the discussion (if needed we could move this discussion to Github Discussions). There's no right or wrong, but it's also far from being arbitrary or just personal choice.

One of the main goals of adding private metadata to vars is to help create public interfaces for consumers. It's a universal concept, independent of programming paradigm. Many (not all) libraries in the Clojure ecosystem judiciously use private metadata to inform consumers what's the public interface and what's not. We don't need to go far, the Clojure source code itself is designed using private vars. Also many tools take advantage of this feature, such as tools to auto-generate docs for Clojure, auto-complete in all major editors, lsp-clojure, and clj-kondo.

Of course we're not designing real libraries in status-mobile, but we do have namespaces which act as if they are *internal libraries*.

On the other hand, in Clojure, **the importance of encapsulating vars is diminished because almost everything is immutable**. And here I say almost, but unfortunately the status-mobile codebase is littered with global, mutable vars (~80 from a quick check).

So having said all that, based on my experience, most of the code in status-mobile will NOT benefit from adding private metadata. I'm also taking in consideration that status-mobile is modified by numerous contributors of varying degrees of experience in Clojure, so I think we should try to keep things simple in our guidelines.

The simplest guideline would be to not use private metadata at all, but that's throwing the baby out with the bathwater.

---

So for the guidelines, here's my draft proposal. I think it's quite sensible for Clojurists.

1. Define atoms and Reagent atoms as private (exceptions are fine).
2. In utils/helper namespaces, define vars as private if they're implementation detail. 
3. Don't use private metadata inside `quo2/` (except for point 1), because the library already has a public interface defined by `quo2.core`.

Let's keep the discussion going o/

Reply:
Very well said @erikseppanen!

> I think we could potentially benefit from more thought put into architecture

Definitely, some topics can't be left for organic growth alone.

> when I think of a library, I think of versioning, breaking changes, semantic versioning, etc.

When you say that I remember a presentation from Rich Hickey where he challenges the notion that semantic versioning is beneficial or that it works. Not a very practical presentation, but quite interesting.

>  I was on a clojure project once where I ran clj-hiera on every build, so developers could always see the latest dependency hierarchy.

I only used `clj-hiera` for pet projects, but what you're describing is super interesting. You reminded now of a tool for Java (https://spring.io/blog/2022/10/21/introducing-spring-modulith) that's capable of automating certain architectural requirements. I'm pretty sure there are other tools in this space. This could be a great complement to the visual capabilities of `clj-hiera`.

> I think in general, with clojure, writing pure functions, pushing out/limiting state/side-effects as much as you can, and maintaining a good namespace hierarchy can go a long way.

Oh yeah, the bread and butter!